plutus-ledger-api/src/PlutusLedgerApi/Common/SerialisedScript.hs

```diff
@@ -56,22 +75,40 @@ scriptCBORDecoder lv pv =
         (p :: UPLC.Program UPLC.FakeNamedDeBruijn DefaultUni DefaultFun ()) <- decodeViaFlat flatDecoder
         pure $ coerce p
 
+-- | The deserialization from a serialised script to a Script (for execution).
+-- Called inside phase-1 validation (assertScriptWellFormed) and inside phase-2's `mkTermToEvaluate`
+fromSerialisedScript :: forall e m. (AsScriptDecodeError e, MonadError e m)
+                     => LedgerPlutusVersion
+                     -> ProtocolVersion
+                     -> SerialisedScript
+                     -> m ScriptForExecution
+fromSerialisedScript lv currentPv sScript = do
+    when (introPv > currentPv)  $
+        throwing _ScriptDecodeError $ LanguageNotAvailableError lv introPv currentPv
+    (remderBS, script) <- deserialiseSScript sScript
+    when (lv /= PlutusV1 && lv /= PlutusV2 && remderBS /= mempty) $
+        throwing _ScriptDecodeError $ RemainderError remderBS
+    pure script
+  where
+    introPv = languageIntroducedIn lv
+    deserialiseSScript :: SerialisedScript -> m (BSL.ByteString, ScriptForExecution)
+    deserialiseSScript = fromShort
+                       >>> fromStrict
```

Comment:
I am not used to `<&>` , and to me `>>>` seems more general. It just the function composition but with args reversed.
```
(<<<) = (.)
(>>>) = flip (.)
``` or sth like that

Reply:
I could have sticked to plain (.) but after a lot of compositions, I lose the track of order. So if the composition points are too many, I reverse the order and go to (>>>). And (>>>) reminds me a lot of the kleisli composition (>=>) that we do use a lot. @michaelpj any stylistic opinion?

Reply:
I think in this instance you're pipelining things so `>>>` seems much better than `<&>` here. I think of `<&>` as `&` inside a functor, and I would need to think about that for a couple of minutes to work out whether it's equivalent to composition...

I do like `>=>` and `<=<`, and it's annoying that there isn't a nice reverse equivalent to `.` except `>>>`, but I think it's clear enough. So :+1: from me.

Reply:
maybe you should add it to the style guide!

Reply:
I was a bit bit confused about what was going on here, but now I understand.  `<&>` works because `(->)` is functorial in its second argument, but I agree with @michaelpj that it's not obvious what's happening.  I was worried about `>>>` because arrows were involved but in fact it's defined in `Control.Category` re-exported from `Control.Arrow`, so arrows are pretty much a red herring.  So I'm willing to buy `>>>` now, but maybe we should import it directly from `Control.Category` to get arrows out of the picture.  An alternative would be to define our own reverse-composition/pipeline opbeerator, but Hoogle shows me that there are already six or seven of those. As @bezirg says, having a reverse-composition/pipeline operator can sometimes make things a lot easier to read, so I'm all in favour of having _some_ standard way to do it.